src/lib.rs

```diff
@@ -156,12 +161,16 @@ pub fn notify(unset_env: bool, state: &[NotifyState]) -> io::Result<()> {
     }
 }
 
+pub unsafe fn notify_and_unset_env(state: &[NotifyState]) -> io::Result<()> {
+    let result = notify(state);
+    unsafe {
```

Comment:
I think older Rust versions throw a warning "unnecessary unsafe" here and we might want to add a warning suppression if we want to support them.

Reply:
What old rust versions are you worried about? Users who depend on this crate will never see a warning because warnings are not shown for dependencies. This crate seems to run with the latest stable in CI. The Rust compiler does not warn for unnecessary `unsafe` here since https://github.com/rust-lang/rust/pull/124636, Rust 1.80 (which would only be relevant for developers, not users of this crate).

Reply:
>What old rust versions are you worried about?

I guess you gave the answer to yourself?

Reply:
This all boils down to the question which MSRV this project supports.
And if this throws a warning with this MSRV, the warning should be fixed IMO.
Otherwise the build logs are cluttered with useless warnings on the test builds that are necessary before release.

Reply:
I don't think it matters. IIRC, when you build a crate, you don't get warnings for its dependencies, so this doesn't affect the users.

As for this crate itself, it's currently missing a `rust-version` declaration (I'm not opposed to adding one). But warnings on the MSRV shouldn't block changes, it targets `stable`.